### DISCUSSION TITLE
Feature request: Load all pages without specifying the number of pages

### DISCUSSION THREAD

**Discussion by jonashaag** [libvips/libvips#2982] [Q&A] [answered]
Eg. in the CLI you can pass `-n 5` but you need to know the `5` first (potentially via vips). If you pass something larger than `5` then it will error out.

**Comment by jcupitt** [comment, marked as answer]
Hi @jonashaag,

You can use -1 to mean all pages, eg.:

```
vips copy huge.pdf[n=-1] x.tif
```

Will write a many page TIFF file. The docs have the details:

https://www.libvips.org/API/current/VipsForeignSave.html#vips-pdfload

**Reply by jonashaag** [reply]
Thanks and sorry that I didn't find that. Maybe we can add that to the CLI list of params?

Like

```
   n            - Load this many pages (-1 = until end of document), input gint
			default: 1
			min: -1, max: 100000
```

**Reply by jcupitt** [reply]
Good idea, I tried to improve the help text: https://github.com/libvips/libvips/commit/d27dfbe55d9ea571ce5bf3fd4948d3804e3ebacc

**Reply by jcupitt** [reply]
... so you now see:

```
$ vips pdfload
load PDF from file
usage:
   pdfload filename out [--option-name option-value ...]
where:
   filename     - Filename to load from, input gchararray
   out          - Output image, output VipsImage
optional arguments:
   page         - First page to load, input gint
			default: 0
			min: 0, max: 100000
   n            - Number of pages to load, -1 for all, input gint
			default: 1
			min: -1, max: 100000
   dpi          - DPI to render at, input gdouble
			default: 72
			min: 0.001, max: 100000
...
```
